dotCMS/src/main/java/com/dotmarketing/db/DbConnectionFactory.java

```diff
@@ -123,6 +133,79 @@ public static DataSource getDataSource() {
         return defaultDataSource;
     }
 
+    /**
+     * Method that loads a datasource from a custom implementation if <b>DATASOURCE_PROVIDER_STRATEGY_CLASS</b>
+     * property is defined. Otherwise, the datasource is initialized using any of these implementations (respecting order):<br>
+     * 1. A db.properties file in WEB-INF/classes implemented by {@link DBPropertiesDataSourceStrategy}<br>
+     * 2. Configuration is taken from environment variables implemented by {@link SystemEnvDataSourceStrategy}<br>
+     * 3. Getting Docker Secrets if set. Implementation: {@link DockerSecretDataSourceStrategy}<br>
+     * 4. A context.xml file in META-INF. Implementation: {@link TomcatDataSourceStrategy}
+     *
+     * @throws InstantiationException
+     * @throws IllegalAccessException
+     * @throws ClassNotFoundException
+     */
+    private static void loadDatasource()
+            throws InstantiationException, IllegalAccessException, ClassNotFoundException {
+        final String providerClassName = Config
+                .getStringProperty("DATASOURCE_PROVIDER_STRATEGY_CLASS", null);
+
+        if (!UtilMethods.isSet(providerClassName)) {
+            if (DBPropertiesDataSourceStrategy.getInstance()
+                    .existsDBPropertiesFile()) {
+                defaultDataSource = DBPropertiesDataSourceStrategy.getInstance()
+                        .apply();
+                Logger.info(DbConnectionFactory.class,
+                        "Datasource loaded from db.properties file");
+            } else if (System.getenv("connection_db_base_url") != null) {
+                defaultDataSource = SystemEnvDataSourceStrategy.getInstance()
+                        .apply();
+                Logger.info(DbConnectionFactory.class,
+                        "Datasource loaded from system environment");
+            } else {
+                defaultDataSource = DockerSecretDataSourceStrategy.getInstance()
+                        .apply();
+                Logger.info(DbConnectionFactory.class,
```

Comment:
if Docker secrets can't be applied, do we log this anyways? might be deceiving

Reply:
shouldn't this `else` check for something like `systemEnvironmentProperties.getVariable("DOCKER_SECRET_FILE_PATH")!=null` like in the Docker strategy, or something else? or if not at least not logging  `Datasource loaded from Docker Secret` in case `defaultDataSource` is null.

Reply:
Done